src/Shared/ResultsHelpers/FileResultHelper.cs

```diff
@@ -52,6 +53,40 @@ internal static async Task WriteFileAsync(HttpContext context, Stream fileStream
         }
     }
 
+    internal static async Task WriteFileAsync(HttpContext context, ReadOnlyMemory<byte> buffer, RangeItemHeaderValue? range, long rangeLength)
+    {
+        var outputStream = context.Response.Body;
+
+        try
+        {
+            if (range == null)
+            {
+                await outputStream.WriteAsync(buffer, context.RequestAborted);
+            }
+            else
+            {
+                var from = 0;
+                var length = 0;
+
+                checked
+                {
+                    // Overflow should throw
+                    from = (int)range.From!.Value;
+                    length = (int)rangeLength;
+                }
+
+                await outputStream.WriteAsync(buffer.Slice(from, length), context.RequestAborted);
+
+            }
+        }
+        catch (OperationCanceledException)
```

Comment:
We recently did this in MVC: https://github.com/dotnet/aspnetcore/pull/39306/files, perhaps we should standardize on it.

```suggestion
        catch (OperationCanceledException) when (context.RequestAborted.IsCancellationRequested)
```

Reply:
Sure, feels unrelated to this PR though (there are other places we need to update).

Reply:
If we do this, there's no need for `context.Abort()`. I think it's best to leave it as-is and not risk an empty 200 response for a canceled operation.